core/src/components/input/test/highlight/index.html

```diff
@@ -174,6 +181,49 @@ <h2>Invalid</h2>
           </div>
         </div>
 
+        <h1>No Fill, Custom</h1>
+        <div class="grid">
+          <div class="grid-item">
+            <h2>Focus</h2>
+            <ion-input
+              label-placement="start"
+              value="hi@ionic.io"
+              class="has-focus custom"
+              label="Email"
+              error-text="Please enter a valid email"
+              helper-text="Enter an email"
+              counter="true"
+              maxlength="20"
+            ></ion-input>
+          </div>
+
+          <div class="grid-item">
+            <h2>Valid, Focus</h2>
+            <ion-input
+              label-placement="start"
+              class="ion-valid has-focus custom"
+              label="Email"
+              error-text="Please enter a valid email"
+              helper-text="Enter an email"
+              counter="true"
+              maxlength="20"
+            ></ion-input>
+          </div>
+
+          <div class="grid-item">
+            <h2>Invalid</h2>
```

Comment:
For some reason, the height isn't being changed for the invalid input when running locally. The same is happening for the others. No issues in the screenshots though.

![Screenshot 2024-03-04 at 4 55 55 PM](https://github.com/ionic-team/ionic-framework/assets/13530427/f5c4fb78-4540-443f-bb29-bc1b0d932310)

Reply:
This is the correct behavior. The invalid styles are only applied when the input has the `has-focus` class:

```
:host(.has-focus) .input-highlight {
  transform: scale(1);
}
```

I am not sure why the other grid items have the `has-focus` class but the invalid one doesn't, but I copied them from the rows above so I didn't want to change what was currently in the test.